### ISSUE TITLE
Grafana_datasource no longer idempotent

### ISSUE THREAD

**Issue by MartyEwings** [voxpupuli/puppet-grafana#270]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet:  7.12.1
- Distribution: centos8
- Module version: 10 

## How to reproduce (e.g Puppet code you use)

called from the metrics dashboard module:

```
  $puppet_metrics_dashboard::influxdb_database_name.each |$db_name| {
    grafana_datasource { "influxdb_${db_name}":
      grafana_url      => "${_uri}://localhost:${puppet_metrics_dashboard::grafana_http_port}",
      type             => 'influxdb',
      database         => $db_name,
      url              => 'http://localhost:8086',
      access_mode      => 'proxy',
      is_default       => false,
      user             => 'admin',
      password         => $puppet_metrics_dashboard::influx_db_password,
      grafana_user     => 'admin',
      grafana_password => $puppet_metrics_dashboard::grafana_password,
      require          => [
        Service['grafana-server'],
        Exec["create influxdb puppet_metrics database ${db_name}"]
      ],
    }
```

## What are you seeing

Since upgrade from 9.0.1  to 10 I'm getting the following when applying a Grafana_datasource resource

/Stage[main]/Puppet_metrics_dashboard::Post_start_configs/Grafana_datasource[influxdb_telegraf]/is_default: is_default changed false to 'false' (corrective)

for some reason we are getting a corrective change from a boolean value to a string, despite boolean being the only thing passed in code

## What behaviour did you expect instead

the boolean value should not be converted to a string

## Output log

/Stage[main]/Puppet_metrics_dashboard::Post_start_configs/Grafana_datasource[influxdb_telegraf]/is_default: is_default changed false to 'false' (corrective)

## Any additional information you'd like to impart

I can see some changes to the resource carried out by rubocop, changing values for is_default from :false to false i don't know if that's the reason

**Comment by alexjfisher** [comment]
@MartyEwings Thanks for the bug report.  You're right in suspecting the recent rubocop autofix commit.  Unfortunately the `rubocop:autocorrect` rake task performs [unsafe corrections](https://github.com/rubocop/rubocop/pull/8192/files#r758505289), and in this case using a symbol instead of a boolean was definitely intended and shouldn't have been changed.
I've proposed https://github.com/voxpupuli/voxpupuli-test/pull/66 and then it shouldn't be difficult for us to revert the breaking parts of https://github.com/voxpupuli/puppet-grafana/commit/70c2adc1abca80bf679f2cd87f72b34e578c912e

**Comment by MartyEwings** [comment]
@alexjfisher thanks for the quick turn around on that one! i don't have to compensate in the metrics dashboard now!

**Comment by root-expert** [comment]
Fixed in 771a032c991efb7e527aad9e23330855d76f22a9 (closing for Changelog)

**Comment by alexjfisher** [comment]
@MartyEwings 10.0.1 is now released and should fix your issue.  Enjoy!
